src/PublishPort.jsx

```diff
@@ -7,10 +8,46 @@
 import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
 import { OutlinedQuestionCircleIcon, TrashIcon } from '@patternfly/react-icons';
 import cockpit from 'cockpit';
+import ipaddr from "ipaddr.js";
+
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import * as utils from './util.js';
 
 const _ = cockpit.gettext;
 
-export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
+export function validatePublishPort(portMapping, key, validationFailed) {
+    const delta = { ...validationFailed };
+
+    switch (key) {
+    case "IP":
+        if (!ipaddr.isValid(portMapping.IP))
+            delta.IP = _("Must be a valid IP address");
+        break;
+    case "hostPort": {
+        if (!utils.isEmpty(portMapping.hostPort)) {
+            const hostPort = parseInt(portMapping.hostPort);
+            if (hostPort < 1 || hostPort > 65536)
+                delta.hostPort = _("Must be between 1 to 65536");
+        }
+
+        break;
+    }
+    case "containerPort": {
+        if (utils.isEmpty(portMapping.containerPort))
+            delta.containerPort = _("Container port must not be empty");
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230907-093411-a9f37bd4-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)